### PR TITLE
chore(backend): abstract pagination support and tests

### LINKED ISSUE #240: Consolidate pagination implementation and tests
There is an increasing number of services with a [pagination method](https://github.com/interledger/rafiki/blob/8efb70b4dd02afec86f8283648840f18bbef9222/packages/backend/src/peer/service.ts#L273).
The implementations and tests are largely redundant and could be consolidated.

### PR DESCRIPTION
**Pull request by wilsonianb**
## Changes proposed in this pull request

- add [custom query builder](https://vincit.github.io/objection.js/recipes/custom-query-builder.html#extending-the-query-builder-in-typescript) to `BaseModel` to support services' `getPage` functionality
- add common `BaseModel` and query resolver pagination tests

## Context

- fixes #240 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/shared/baseModel.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      }
+      const modelsBackwards = await getPage(paginationBackwards)
+      expect(modelsForwards).toHaveLength(10)
+      expect(modelsBackwards).toHaveLength(10)
+      expect(modelsForwards).toEqual(modelsBackwards)
+    })
+  })
+}
+
+test.todo('test suite must contain at least one test')
```
Comment (sentientwaffle) [inline packages/backend/src/shared/baseModel.test.ts:80]:
```suggestion
```

**Review by sentientwaffle — APPROVED** [review]
:tada:

File: packages/backend/src/shared/baseModel.test.ts (hunk 1 above)
Comment (wilsonianb) [reply packages/backend/src/shared/baseModel.test.ts:80]: Tests fail without that.
The file _could_ be excluded from tests either via `jest.config.js` or changing the `.test.ts` extension

Comment (sentientwaffle) [reply packages/backend/src/shared/baseModel.test.ts:80]: Ah, nevermind, I didn't think of that.

**Pull request merged**